src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -147,6 +147,18 @@ public DataFrame(params DataFrameColumn[] columns)
             }
         }
 
+        /// <summary>
+        /// An indexer based on <see cref="DataFrameColumn.Name"/>
+        /// </summary>
+        /// <param name="columnName">The name of a <see cref="DataFrameColumn"/></param>
+        /// <returns>A <see cref="DataFrameColumn"/> if it exists.</returns>
+        /// <exception cref="ArgumentException">Throws if <paramref name="columnName"/> is not present in this <see cref="DataFrame"/></exception>
+        public DataFrameColumn this[string columnName]
+        {
+            get => Columns[columnName];
```

Comment:
The case that the column doesn't exist is handled and tested.
What's about the case of `columnName = null`?

Reply:
As it turns out, we have too many overloads of this indexer, so the only way to call this is to do `df[(string)null]`. I fixed the null case just in case though :)